app/controllers/AddAdditionalMetadataController.scala

```diff
@@ -34,248 +34,104 @@ class AddAdditionalMetadataController @Inject() (
-        cache.set(s"$consignmentId-reference", consignment.consignmentReference, 1.hour)
-        // Set the values to those of the first file's metadata until we decide what to do with multiple files.
-        updateFormFields(defaultFieldsForForm, consignment.files.headOption.map(_.fileMetadata).getOrElse(Nil))
-      }
-    } yield {
-      val files: List[File] = getFilesFromConsignment(consignment.files)
-      // Call to details.parentFolderId.get should be temporary. User shouldn't see this page if the parent ID is empty.
-      val pageInfo = PageInfo(request.token.name, consignment.consignmentReference, "", "", updatedFieldsForForm)
-      val controllerInfo = ControllerInfo(metadataType, isMainForm, fieldsAndValuesSelectedOnPrevPage, consignmentId, files)
-      (pageInfo, controllerInfo)
+  private def buildUpdateMetadataInput(updatedFormFields: List[FormField]): List[UpdateFileMetadataInput] = {
+    updatedFormFields.flatMap {
+      case TextField(fieldId, _, _, multiValue, nameAndValue, _, _, _) =>
+        UpdateFileMetadataInput(filePropertyIsMultiValue = multiValue, fieldId, nameAndValue.value) :: Nil
+      case DateField(fieldId, _, _, multiValue, day, month, year, _, _, _) =>
+        val dateTime: LocalDateTime = LocalDate.of(year.value.toInt, month.value.toInt, day.value.toInt).atTime(LocalTime.MIDNIGHT)
+        UpdateFileMetadataInput(filePropertyIsMultiValue = multiValue, fieldId, Timestamp.valueOf(dateTime).toString) :: Nil
+      case RadioButtonGroupField(fieldId, _, _, _, multiValue, _, selectedOption, _, _, dependencies, _) =>
+        val fileMetadataInputs = buildUpdateMetadataInput(dependencies.values.flatten.toList)
+        UpdateFileMetadataInput(filePropertyIsMultiValue = multiValue, fieldId, stringToBoolean(selectedOption).toString) :: fileMetadataInputs
+      case DropdownField(fieldId, _, _, multiValue, _, selectedOption, _, _) =>
+        UpdateFileMetadataInput(filePropertyIsMultiValue = multiValue, fieldId, selectedOption.map(_.value).getOrElse("")) :: Nil
     }
-
-  private def getValuesThatWereSelectedIfTheyHaveDependencies(
-      dependencyProperties: Set[CustomMetadata],
-      metadataInput: List[UpdateFileMetadataInput]
-  ): Set[ValueSelectedAndDepsToDel] = {
-    val propertiesThatHaveValuesWithDependencies: Set[CustomMetadata] = getPropertiesWhereValuesHaveDependencies(dependencyProperties)
-    val valuesThatWereSelectedThatHaveDependencies: Set[ValueSelectedAndDepsToDel] = getValueSelectedIfItHasDependencies(metadataInput, propertiesThatHaveValuesWithDependencies)
-    valuesThatWereSelectedThatHaveDependencies
   }
 
-  private def getDefaultFieldsForForm(
-      metadataType: String,
-      isMainForm: Boolean,
-      staticMetadata: Set[StaticMetadata],
-      consignmentId: UUID,
-      request: Request[AnyContent]
-  ): Future[FormData] = {
+  private def getDefaultFieldsForForm(consignmentId: UUID, request: Request[AnyContent]): Future[List[FormField]] = {
```

Comment:
Maybe rename this: `getFieldsForForm`, as there is only a single page, so the fields aren't really "default"